### ISSUE TITLE
Pundit role on an argument does not restrict access

### ISSUE THREAD

**Issue by gopeter** [rmosolgo/graphql-ruby#3384]
Dear @rmosolgo,

this issue is based on #3184, but about another bug I may have found. 

I tried to dive again into #3184 and found out, that the arguments are resolved now? Using `argument :is_admin, Boolean, required: false, pundit_role: :can_set_admin` and the following policy ...

```ruby
def can_set_admin?
  puts record.arguments[:id].id
  puts user.id
  puts user.admin? && record.arguments[:id].id != user.id

  user.admin? && record.arguments[:id].id != user.id
end
```

... I get this:

```ruby
1
1
false
```

But I can execute the mutation without any errors. Even if I do ...

```ruby
def can_set_admin?
  false
end
```

... I can set this argument in the mutation without any errors. Is there anything new that I've missed?

This is my `BaseMutation` and I'm using `graphql 1.12.5` and `graphql-pro 1.17.9`:

```ruby
module Mutations
  class BaseMutation < GraphQL::Schema::RelayClassicMutation
    include GraphQL::Pro::PunditIntegration::MutationIntegration

    object_class Types::BaseMutationPayload
    field_class Types::BaseField
    argument_class Types::BaseArgument

    field :errors, Types::Scalars::Json, null: true

    def unauthorized_by_pundit(owner, value)
      # Return errors as data:
      if value.is_a?(ActiveRecord::Base)
        { errors: ["Missing required permission: `#{owner.pundit_role}` for #{value.model_name} with ID #{value.id}"] }
      else
        { errors: ["Missing required permission: `#{owner.pundit_role}`"] }
      end
    end
  end
end
```

**Comment by rmosolgo** [comment]
Hey, thanks for the detailed bug report! 

I think a bug was introduced when I refactored argument authorization a while back. After that, the argument authorization codepath for the Pundit integration didn't properly wrap the `unauthorized_by_pundit?` result with `[false, result]` to support errors-as-data. I didn't have test coverage for that _specific_ path, so I'm going to look into a test coverage assessment framework to make sure that future refactors don't reduce my test coverage! 

Sorry for the trouble. I've just released graphql-pro 1.17.11 which should address this issue. Could you give it a try and let me know if you run into any more trouble?

**Comment by gopeter** [comment]
Dear @rmosolgo,

thanks, works like a charm!
